core_java/grep/src/main/java/ca/jrvs/apps/grep/JavaGrepLambdaImp.java

```diff
@@ -0,0 +1,68 @@
+package ca.jrvs.apps.grep;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.apache.log4j.BasicConfigurator;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.log4j.BasicConfigurator;
+
+public class JavaGrepLambdaImp extends javaGrepImp {
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("Usage: JavaGrep regex rootPath outFile");
+    }
+    BasicConfigurator.configure();
+    JavaGrepLambdaImp javaGrepLambdaImp = new JavaGrepLambdaImp();
+    javaGrepLambdaImp.setRegex(args[0]);
+    javaGrepLambdaImp.setRootPath(args[1]);
+    javaGrepLambdaImp.setOutFile(args[2]);
+    try {
+      javaGrepLambdaImp.process();
+    } catch (Exception ex) {
+      javaGrepLambdaImp.logger.error(ex.getMessage(), ex);
+    }
+  }
+
+  @Override
+
+  public List<File> listFiles(String rootDir) {
+    try {
+      Stream<File> filenames = Files.walk(Paths.get(rootDir))
+          .filter(Files::isRegularFile)
+          .map(Path::toFile);
+
+      return filenames.collect(Collectors.toList());
+    } catch (IOException e) {
+      logger.error("IOException", e);
+      throw new IllegalArgumentException("IOException");
```

Comment:
throw new IllegalArgumentException("IOException");
write a app-specific and meaningful message